extensions/azurecore/src/account-provider/tokenCache.ts

```diff
@@ -223,22 +223,16 @@ export default class TokenCache implements adal.TokenCache {
 		return this.getOrCreateEncryptionParams()
 			.then(encryptionParams => {
 				try {
-					let cacheCipher = fs.readFileSync(self._cacheSerializationPath, TokenCache.FsOptions);
-
-					let decipher = crypto.createDecipheriv(TokenCache.CipherAlgorithm, encryptionParams.key, encryptionParams.initializationVector);
-					let cacheJson = decipher.update(cacheCipher, 'hex', 'binary');
-					cacheJson += decipher.final('binary');
-
-					// Deserialize the JSON into the array of tokens
-					let cacheObj = <adal.TokenResponse[]>JSON.parse(cacheJson);
-					for (let objIndex in cacheObj) {
-						// Rehydrate Date objects since they will always serialize as a string
-						cacheObj[objIndex].expiresOn = new Date(<string>cacheObj[objIndex].expiresOn);
-					}
-
-					return cacheObj;
+					return self.decryptCache('utf8', encryptionParams);
 				} catch (e) {
-					throw e;
+					try {
+						// try to parse using 'binary' encoding and rewrite cache as UTF8
+						let response = self.decryptCache('binary', encryptionParams);
+						self.writeCache(response);
+						return response;
+					} catch (e) {
+						throw e;
```

Comment:
this inner try/catch isn't necessary since it doesn't do anything except rethrow `e`

Reply:
yeah, just following the pattern of the rest of the code in this file which does this for some reason.